apis/apiextensions/v1alpha1/config_set_types.go

```diff
@@ -0,0 +1,33 @@
+package v1alpha1
+
+import (
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +genclient
+// +genclient:nonNamespaced
+
+// A ConfigSet contains a set of arbitrary, unstructured values.
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:resource:scope=Cluster,categories=crossplane
+type ConfigSet struct {
```

Comment:
I'd like to call this an `EnvironmentConfig` rather than a `ConfigSet`, for symmetry with our existing `ProviderConfig` and `StoreConfig` types. Or more generally I'd like to call it something that ends in `Config` - I'm happy if there are better suggestions than `Environment`.

Reply:
What about `CompositionEnvironmentConfig`?

Reply:
Solved in https://github.com/crossplane/crossplane/pull/3008#discussion_r902352449